js/components/closed-caption-multi-audio-menu/helpers.js

```diff
@@ -90,13 +88,75 @@ function getDisplayTitle(language, label) {
   } else {
     return displayLanguage.concat(' ', displayLabel);
   }
+}
+
+/**
+ * Transforms tracks list based on criteria
+ * if all tracks are distinct - only use language attribute
+ * if there are duplicates - append label to the language attribute
+ * @param {Array} tracksList - list of all tracks
+ * @returns {Array} transformedTracksList - list of transformed tracks
+ */
+function transformTracksList(tracksList) {
+  var transformedTracksList = [];
+  // first we group by language to know if we have distinct tracks
+  if (tracksList && tracksList.length) {
+    var groupedTracks = _.groupBy(tracksList, 'language');
+    var groupedKeys = _.keys(groupedTracks);
+
+    // if all languages are distinct - discard labels
+    if (groupedKeys.length === tracksList.length) {
+      transformedTracksList = tracksList.map(function(audioTrack) {
+        var trackDisplayTitle = getDisplayTitle(audioTrack.language);
+        var transformedTrack = {
+          id: audioTrack.id,
+          label: trackDisplayTitle,
+          enabled: audioTrack.enabled
+        };
+
+        return transformedTrack;
+      });
+    } else {
+      var uniqueTracks = groupedKeys.map(function(key) {
+        // if there are multiple tracks with the same language code
+        if (groupedTracks[key].length > 1) {
+          // get each list of duplicating tracks
+          return groupedTracks[key].map(function(audioTrack) {
```

Comment:
It looks like we could possibly move this mapping to its own function, since it's used several times.